src/views/CreateGroup/components/CreateGroupForm/CreateGroupForm.tsx

```diff
@@ -0,0 +1,177 @@
+import React, { FC, useState } from 'react';
+import { makeStyles } from '@material-ui/styles';
+import TextField from '@material-ui/core/TextField';
+import { Theme, Grid, Typography } from '@material-ui/core';
+import Fade from '@material-ui/core/Fade';
+import Button from '@material-ui/core/Button';
+import Collapse from '@material-ui/core/Collapse';
+import Card from '@material-ui/core/Card';
+import GroupRoundedIcon from '@material-ui/icons/GroupRounded';
+import GroupTypeControl from './GroupTypeControl';
+import { Formik, Form } from 'formik';
+import validationSchema from './validationSchema';
+import PasswordInput from './PasswordInput';
+import GroupSettingsControl from './GroupSettingsControl';
+import { initialValues, CreateGroupFormValues } from './formInitialValues';
+
+const useStyles = makeStyles((theme: Theme) => ({
+	card: {
+		padding: theme.spacing(1),
+		marginTop: theme.spacing(4),
+		display: 'flex',
+		flexDirection: 'column',
+	},
+	inputsContainer: {
+		display: 'flex',
+		flexDirection: 'column',
+		padding: theme.spacing(4),
+	},
+	actionButton: {
+		marginTop: theme.spacing(4),
+		width: '35%',
+		[theme.breakpoints.down('lg')]: {
+			width: '100%',
+		},
+		alignSelf: 'center',
+	},
+	groupImageContainer: {
+		display: 'flex',
+		flex: 1,
+		flexDirection: 'column',
+		justifyContent: 'space-around',
+		alignItems: 'center',
+	},
+	groupImage: {
+		height: 200,
+		width: 200,
+		borderRadius: '50%',
+		objectFit: 'cover',
+	},
+	uploadImage: {
+		marginTop: theme.spacing(4),
+		marginBottom: theme.spacing(4),
+
+		background: theme.palette.primary.main,
+		padding: theme.spacing(1),
+		color: 'white',
+		borderRadius: 4,
+		cursor: 'pointer',
+	},
+}));
+
+interface Props {
+	submitForm: (body: CreateGroupFormValues, groupAvatar: string) => void;
+}
+const CreateGroupForm: FC<Props> = ({ submitForm }) => {
+	const classes = useStyles();
+	const [groupImage, setGroupImage] = useState('');
+
+	const handleImageSelection = async (e: any) => {
+		const file = e.target.files[0];
+		if (file) {
+			const reader = new FileReader();
+			reader.onloadend = () => {
+				const b64 = reader.result;
+				if (typeof b64 === 'string') setGroupImage(b64);
+			};
+			reader.readAsDataURL(file);
+		}
+	};
+	return (
+		<Formik
```

Comment:
Don't change it now - but FYI - we already have a validation library being used in the project (validate.js). 
It probably was better to use it instead of adding another one.
but it is ok for now to continue with it

Reply:
Actually i saw the validate.js library, the issue is that its schema is not a valid schema when using Formik.
The benefit we get by using formik is quite big so adding yup for the validation schema seemed reasonable for me. If you still dont like the idea of 2 validation packages let me know and ill try to find a better solution (maybe even replace validate.js in the auth screens to Yup).